assets/js/blocks-registry/payment-methods/payment-method-config-helper.js

```diff
@@ -8,3 +8,39 @@ export const canMakePaymentWithFeaturesCheck = ( canMakePayment, features ) => (
 	);
 	return featuresSupportRequirements && canMakePayment( canPayArgument );
 };
+
+// Filter out payment methods by callbacks registered by extensions
+export const canMakePaymentWithExtensions = (
+	canMakePayment,
+	extensionsCallbacks,
+	paymentMethodName
+) => ( canPayArgument ) => {
+	// validate whether the payment method is available
+	let canPay = canMakePayment( canPayArgument );
+
+	if ( canPay ) {
+		const namespacedCallbacks = {};
+		Object.entries( extensionsCallbacks ).forEach(
+			( [ namespace, callbacks ] ) => {
+				if ( typeof callbacks[ paymentMethodName ] === 'function' ) {
```

Comment:
I think this check should be part of the registration so it won't be necessary here